src/plugins/workspace/server/saved_objects/saved_objects_wrapper_for_check_workspace_conflict.ts

```diff
@@ -0,0 +1,257 @@
+          } else {
+            savedObjectWorkspaces = currentItem.workspaces;
+          }
+        }
+      }
+
+      return await wrapperOptions.client.create(type, attributes, {
+        ...options,
+        workspaces: savedObjectWorkspaces,
+      });
+    };
+
+    const bulkCreateWithWorkspaceConflictCheck = async <T = unknown>(
+      objects: Array<SavedObjectsBulkCreateObject<T>>,
+      options: SavedObjectsCreateOptions = {}
+    ): Promise<SavedObjectsBulkResponse<T>> => {
+      const { overwrite, namespace } = options;
+      const bulkGetDocs = objects
+        .filter((object) => !!(object.id && options.workspaces))
+        .map((object) => {
+          const { type, id } = object;
+          /**
+           * It requires a check when overwriting objects to target workspaces
+           */
+          return {
+            type,
+            id: id as string,
+            fields: ['id', 'workspaces'],
+          };
+        });
+      const objectsConflictWithWorkspace: SavedObject[] = [];
+      const objectsMapWorkspaces: Record<string, string[] | undefined> = {};
+      if (bulkGetDocs.length) {
+        const bulkGetResult = await wrapperOptions.client.bulkGet(bulkGetDocs);
+
+        bulkGetResult.saved_objects.forEach((object) => {
+          if (!object.error && object.id && overwrite) {
+            /**
+             * When it is about to overwrite a object into options.workspace.
+             * We need to check if the options.workspaces is the subset of object.workspaces,
+             * Or it will be treated as a conflict
+             */
+            const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToSourceWorkspaces(
+              options.workspaces,
+              object.workspaces
+            );
+            const { id, type } = object;
+            if (filteredWorkspaces.length) {
+              /**
+               * options.workspaces is not a subset of object.workspaces,
+               * return a conflict error.
+               */
+              objectsConflictWithWorkspace.push({
+                id,
+                type,
+                attributes: {},
+                references: [],
+                error: {
+                  ...errorContent(SavedObjectsErrorHelpers.createConflictError(type, id)),
+                  metadata: { isNotOverwritable: true },
+                },
+              });
+            } else {
+              objectsMapWorkspaces[this.getRawId({ namespace, type, id })] = object.workspaces;
+            }
+          }
+        });
+      }
+
+      const objectsFilteredByError = objects.filter(
+        (item) =>
+          !objectsConflictWithWorkspace.find(
+            (errorItems) =>
+              this.getRawId({ namespace, type: errorItems.type, id: errorItems.id }) ===
+              this.getRawId({ namespace, type: item.type, id: item.id as string })
+          )
+      );
+      let objectsPayload = objectsFilteredByError;
```

Comment:
`objectsFilteredByError` is useless, how about filter by `objectsMapWorkspaces` and map the workspaces.

Reply:
Sure, done for that.